spring-integration-core/src/main/java/org/springframework/integration/handler/advice/RequestHandlerCircuitBreakerAdvice.java

```diff
@@ -95,9 +95,10 @@ private AtomicInteger getFailures() {
 
 		private static final long serialVersionUID = 1L;
 
-		public CircuitBreakerOpenException(String message) {
+		private CircuitBreakerOpenException(String message) {
```

Comment:
Good point; I blindly corrected the Sonar complaints. I don't think it needs to be a top-level class, but it should be `public static...` - the `AdvisedMetadata` should be `static` too (but `private`).